docker/compose/swarm/scale_segmentstore

```diff
@@ -17,7 +17,7 @@ try:
         "docker service inspect pravega_segmentstore --format='{{.Spec.TaskTemplate.ContainerSpec.Env}}'", shell=True)
 
     def get_config(name):
-        return re.findall(name + "=(.+?)( |])", env)[0][0]
+        return re.findall(name + "=(.+?)( |]|\r\n|\n)", env)[0][0]
```

Comment:
One clarification question about the second part of the regular expression: `( |]|\r\n|\n)`. The first part is making sure that after the `=` sign there is a string with one or more `.`.  The second part is a disjunction with four options:

1- Space
2- Closing bracket `]`
3- End of line `\r\n`
4- Line feed `\n`

I suppose that (1) is for the case the user accidentally leaves a space after the address. I'm not sure about (2), though. (3) and (4) make sense.

Could you clarify (1) and (2)?

Reply:
The output [1] of `docker service inspect` command executed by the `scale_segmentstore` can include `]`   e.g: `ZK_URL=zookeeper:2181]` . 
An example of the space included is `HDFS_URL=hdfs://hdfs:8020 `.

```
docker service inspect pravega_segmentstore --format='{{.Spec.TaskTemplate.ContainerSpec.Env}}'
[CONTROLLER_URL=tcp://controller:9090 HDFS_REPLICATION=1 HDFS_URL=hdfs://hdfs:8020 JAVA_OPTS=-Dmetrics.enableCSVReporter=false
-Dpravegaservice.publishedIPAddress=10.0.2.15
-Dpravegaservice.listeningIPAddress=0
-Dbookkeeper.bkEnsembleSize=2
-Dbookkeeper.bkAckQuorumSize=2
-Dbookkeeper.bkWriteQuorumSize=2
-Xmx900m
-XX:OnError="kill -9 p%"
-XX:+ExitOnOutOfMemoryError
-XX:+CrashOnOutOfMemoryError
-XX:+HeapDumpOnOutOfMemoryError
 TIER2_STORAGE=HDFS WAIT_FOR=bookkeeper:3181,hdfs:8020 ZK_URL=zookeeper:2181]
```